Add parsing of ENV variable values to Boolean type

mjgiarlo opened the pull request:
Fixes #178 

(Discovered in https://github.com/samvera-labs/hyku/issues/1462. Heads-up, @tdonohue.)

mjgiarlo commented:
@pkuczynski I can eliminate the CodeClimate issue by taking the logic from `#__boolean` and inlining it in `#__value`, but I didn't want to change the one-liner pattern going there. What do you advise? Should I make CodeClimate happy, or are there other changes you'd like to see?

pkuczynski commented:
I would rather see this done as:

```ruby
# Try to convert string to a correct type
    def __value(v)
      case value
      when 'false'
        false
      when 'true'
        true
      else
        Integer(v) rescue Float(v) rescue v
      end
    end
```

mjgiarlo commented on lib/config/options.rb:
```diff
@@ -188,7 +188,14 @@ def __convert(h) #:nodoc:
     # Try to convert string to a correct type
     def __value(v)
-      Integer(v) rescue Float(v) rescue v
+      case value
```
`s/value/v/` here, right?

pkuczynski replied:
yeah, totally! sorry!

mjgiarlo commented:
@pkuczynski I fixed a typo in the change you submitted, squashed commits, and pushed again. Thanks for the change! I suspect this is now ready to go.

When do you think 1.6.0 can get released with this change?

pkuczynski commented:
Done! Released as `1.5.1`.

pkuczynski commented:
Thanks for your contributiuon :)

mjgiarlo commented:
Thanks for the useful gem, @pkuczynski!
